pkg/queue/sharedmain/handlers.go

```diff
@@ -145,3 +162,29 @@ func withFullDuplex(h http.Handler, enableFullDuplex bool, logger *zap.SugaredLo
 		h.ServeHTTP(w, r)
 	})
 }
+
+func isProbeRequest(r *http.Request) bool {
+	// Check standard probes (K8s and Knative probe headers)
+	if netheader.IsProbe(r) {
+		return true
+	}
+
+	// Check all Knative internal probe user agents that should not be counted
+	// as pending requests (matching what the Drainer filters)
+	userAgent := r.Header.Get("User-Agent")
+	return strings.HasPrefix(userAgent, netheader.ActivatorUserAgent) ||
+		strings.HasPrefix(userAgent, netheader.AutoscalingUserAgent) ||
+		strings.HasPrefix(userAgent, netheader.QueueProxyUserAgent) ||
+		strings.HasPrefix(userAgent, netheader.IngressReadinessUserAgent)
```

Comment:
I thought the autoscaler and activator etc all use [`K-Network-Probe`](https://github.com/knative/networking/blob/8cc248b445a0575054d8a9b551de39a93d4fe267/pkg/http/header/header.go#L41) header which is covered by `netheader.IsProbe(r)`

Do you know what exactly is getting through here?

Reply:
not sure exactly, https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_serving/16080/upgrade-tests_serving_main/1973128182416543744 is complaining about it. in practice, I have not observed a problem